sys/posix/pthread/pthread_spin.c

```diff
@@ -47,7 +47,7 @@ int pthread_spin_lock(pthread_spinlock_t *lock)
         return EINVAL;
     }
 
-    while (atomic_set_to_one((int *)lock) == 0) {
+    while (atomic_set_to_one((atomic_int_t *)lock) == 0) {
```

Comment:
Somehow I don't like this typecast but I can't really say why. Maybe change the def to:
```c
typedef struct {
    atomic_int_t value;
} pthread_spinlock_t;
```
?